internal/interpreter/api.rs

```diff
@@ -668,6 +668,17 @@ impl ComponentDefinition {
         })
     }
 
+    /// Creates a new instance of the component and returns a shared handle to it.
+    pub fn create_embedded(&self, ctx: FactoryContext) -> Result<ComponentInstance, PlatformError> {
```

Comment:
Need to be marked as #[doc(hidden)] and properly gated. 
We do not want that in the public API at this point

Reply:
I think here `#[doc(hidden)] ` should actually be enough since `FactoryContext` is already not creatable.

(But having an internal cargo feature would make it even more gated)